### PR TITLE
PUBDEV-7069 - Protection against non-valid(BAD) types in DKVClearTask

### PR DESCRIPTION
**Pull request by Pscheidl**
DO NOT MERGE - Custom code for our customer.

https://0xdata.atlassian.net/projects/PUBDEV/issues/PUBDEV-7069?filter=myopenissues&orderby=priority%20DESC

### CODE REVIEW

File: h2o-core/src/main/java/water/DKVManager.java [hunk 1]
```diff
@@ -111,10 +112,17 @@ protected void setupLocal() {
         }
         if(value.isNull()) continue;
 
-        if (value.isFrame()) {
-          ((Frame) value.get()).retain(removalFutures, retainedKeys);
-        } else if (value.isModel()) {
-          ((Model)value.get()).remove(removalFutures);
+        try {
+          TypeMap.getTheFreezableOrThrow(value.type());
```
Comment (Pscheidl) [inline h2o-core/src/main/java/water/DKVManager.java]: It declares ClassNotFoundException and makes the catch block possible to compile. The exception itself may occur in the `value.isFrame()` or `value.isModel` calls below as well.

**Comment by Pscheidl** [comment]
I'll try to merge this code once a reproducible test case is found.

File: h2o-core/src/main/java/water/DKVManager.java [hunk 2]
```diff
@@ -111,10 +112,17 @@ protected void setupLocal() {
...
+          if (value.isFrame()) {
+            ((Frame) value.get()).retain(removalFutures, retainedKeys);
+          } else if (value.isModel()) {
+            ((Model) value.get()).remove(removalFutures);
+          }
+
+        } catch (ClassNotFoundException e) {
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/DKVManager.java]: Please add a comment on why is this exception caught.

Comment (Pscheidl) [reply h2o-core/src/main/java/water/DKVManager.java]: Done-.

**Comment by Pscheidl** [comment]
Also added a little bit more JavaDoc.

**Pull request merged**